packages/shared/src/schemas.ts

```diff
@@ -0,0 +1,4 @@
+import { z } from 'zod';
+
+export const ETH_ADDRESS_REGEXP = new RegExp('^0x[a-fA-F0-9]{40}$');
+export const EthAddressSchema = z.string().regex(ETH_ADDRESS_REGEXP);
```

Comment:
Should we go a step further and validate the address with ethers and a zod `.refine`?

https://docs.ethers.org/v6/api/address/#getAddress